include/secp256k1_musig.h

```diff
@@ -375,113 +180,337 @@ SECP256K1_API int secp256k1_musig_partial_signature_serialize(
- *  In:   partial_sig: signature to verify (cannot be NULL)
- *             pubkey: public key of the signer who produced the signature (cannot be NULL)
+ *  who produced an invalid signature.
+ *
+ *  Returns: 0 if the arguments are invalid or the partial signature does not
+ *           verify, 1 otherwise
+ *  Args         ctx: pointer to a context object, initialized for verification
+ *  In:  partial_sig: pointer to partial signature to verify
+ *          pubnonce: public nonce sent by the signer who produced the signature
+ *            pubkey: public key of the signer who produced the signature
+ *      keyagg_cache: pointer to the keyagg_cache that was output when the
+ *                    aggregate public key for this session
+ *           session: pointer to the session that was created with
+ *                    musig_nonce_process
  */
 SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_musig_partial_sig_verify(
     const secp256k1_context* ctx,
-    const secp256k1_musig_session *session,
-    const secp256k1_musig_session_signer_data *signer,
-    const secp256k1_musig_partial_signature *partial_sig,
-    const secp256k1_xonly_pubkey *pubkey
-) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4) SECP256K1_ARG_NONNULL(5);
+    const secp256k1_musig_partial_sig *partial_sig,
+    const secp256k1_musig_pubnonce *pubnonce,
+    const secp256k1_xonly_pubkey *pubkey,
+    const secp256k1_musig_keyagg_cache *keyagg_cache,
+    const secp256k1_musig_session *session
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4) SECP256K1_ARG_NONNULL(5) SECP256K1_ARG_NONNULL(6);
 
-/** Combines partial signatures
- *
- *  Returns: 1: all partial signatures have values in range. Does NOT mean the
- *              resulting signature verifies.
- *           0: some partial signature are missing or had s or r out of range
- *  Args:         ctx: pointer to a context object (cannot be NULL)
- *            session: initialized session for which the combined nonce has been
- *                     computed (cannot be NULL)
- *  Out:        sig64: complete signature (cannot be NULL)
- *  In:  partial_sigs: array of partial signatures to combine (cannot be NULL)
- *             n_sigs: number of signatures in the partial_sigs array
+/** Aggregates partial signatures
+ *
+ *  Returns: 0 if the arguments are invalid, 1 otherwise (which does NOT mean
+ *           the resulting signature verifies).
+ *  Args:         ctx: pointer to a context object
+ *  Out:        sig64: complete (but possibly invalid) Schnorr signature
+ *  In:       session: pointer to the session that was created with
+ *                     musig_nonce_process
+ *       partial_sigs: array of pointers to partial signatures to aggregate
+ *             n_sigs: number of elements in the partial_sigs array
```

Comment:
Nit, follow up to https://github.com/ElementsProject/secp256k1-zkp/pull/131#discussion_r770954273, sorry! 😓

```suggestion
 *             n_sigs: number of elements in the partial_sigs array. Must be
 *                     greater than 0.
```